bitcoin/src/blockdata/transaction.rs

```diff
@@ -1065,23 +1071,29 @@ impl InputWeightPrediction {
     /// # Returns
     ///
     /// `InputWeightPrediction` with `input_script_len` and `witness_element_lengths` calculated.
+    /// Both aforementioned variables will saturate to u32::MAX during calculation.
     pub fn new<T>(input_script_len: usize, witness_element_lengths: T) -> Self
     where
         T: IntoIterator,
         T::Item: Borrow<usize>,
     {
         let (count, total_size) = witness_element_lengths.into_iter().fold(
-            (0usize, 0),
+            (0usize, 0u32), // saturates to (_, u32::MAX)
             |(count, total_size), elem_len| {
                 let elem_len = *elem_len.borrow();
                 let elem_size = Self::add_to_encoded_size(elem_len);
-                (count + 1, total_size + elem_size)
+                (count + 1, total_size.saturating_add(elem_size))
             },
         );
-        let witness_size =
-            if count > 0 { total_size + compact_size::encoded_size(count) } else { 0 };
+
+        // cast ok, range of encoded_size is 0..0xFFFFFFFF
+        let encoded_size = compact_size::encoded_size(count).try_into().unwrap();
+        let witness_size = if count > 0 { total_size.saturating_add(encoded_size) } else { 0 };
+
         let script_size = Self::add_to_encoded_size(input_script_len);
 
+        let script_size = script_size.try_into().unwrap(); // cast ok, u32 fits surpported usize
+        let witness_size = witness_size.try_into().unwrap(); // cast ok, u32 fits surpported usize
```

Comment:
In 16caed073dc34842ab9030f27ea4b06c5e181b71:

typo `surpported`

Reply:
done